repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphRetriever.java

```diff
@@ -1093,6 +1178,105 @@ private AtlasEntityHeader mapVertexToAtlasEntityHeader(AtlasVertex entityVertex,
         return ret;
     }
 
+    private AtlasEntityHeader mapVertexToAtlasEntityHeaderWithPrefetch(AtlasVertex entityVertex, Set<String> attributes) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("mapVertexToAtlasEntityHeaderWithPrefetch");
+        AtlasEntityHeader ret = new AtlasEntityHeader();
+        try {
+            //pre-fetching the properties
+            String typeName = entityVertex.getProperty(Constants.TYPE_NAME_PROPERTY_KEY, String.class); //properties.get returns null
+            AtlasEntityType entityType = typeRegistry.getEntityTypeByName(typeName); // this is not costly
+            Map<String, Object> properties = preloadProperties(entityVertex, entityType, attributes);
+
+            String guid = (String) properties.get(Constants.GUID_PROPERTY_KEY);
+
+            Integer value = (Integer)properties.get(Constants.IS_INCOMPLETE_PROPERTY_KEY);
+            Boolean isIncomplete = value != null && value.equals(INCOMPLETE_ENTITY_VALUE) ? Boolean.TRUE : Boolean.FALSE;
+
+            ret.setTypeName(typeName);
+            ret.setGuid(guid);
+
+            String state = (String)properties.get(Constants.STATE_PROPERTY_KEY);
+            Id.EntityState entityState = state == null ? null : Id.EntityState.valueOf(state);
+            ret.setStatus((entityState == Id.EntityState.DELETED) ? AtlasEntity.Status.DELETED : AtlasEntity.Status.ACTIVE);
+
+            RequestContext context = RequestContext.get();
+            boolean includeClassifications = context.includeClassifications();
+            boolean includeClassificationNames = context.isIncludeClassificationNames();
+            if(includeClassifications){
+                ret.setClassificationNames(getAllTraitNamesFromAttribute(entityVertex));
```

Comment:
We are setting the same values in 'if" and "else"

Reply:
Please verify.

Reply:
Thats snippet from old code, just did copy paste